### PR TITLE
Feature/get active combinations

### PR DESCRIPTION
**Pull request by kapppa-joe**
- Add a function to select the active Category/MarketSector combinations from data
- Add a a pipeline job script to generate above list and save to blob storage
- Update README.md about pipeline jobs in this repo

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
...
-1. TODO: Write about pipeline
+-------------------------------------------------------
 
 ### Testing
 The unit test of this repo is located in `test` directory. You can use the below command in a terminal to run the full \
 test suite:
 
 ```shell
-python3 -m unittest discover -p '*_test.py' -s test
+pytest .
```
Comment (charlietw) [inline README.md]: Think this would need to be `poetry run pyest` wouldn't it?

Or even better -  `make unit_testing`

File: README.md [hunk 2]
```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
    should be analysed qualitatively by the data scientist for each category & sector (there are over 150 different 
    combination) before finalising the best value.
 
+-------------------------------------------------------
+### Data pipeline
```
Comment (charlietw) [inline README.md:106]: This section is absolutely brilliant, great work.

File: README.md [hunk 3]
```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
...
+This job insert dummy records of amount £0 spending into spend data for in-between months that has no spending.
+
+For every Category/Market sector combinations, it will find the earliest and most recent dates of spend record, and within the period if there is any month that doesn't have any record, it will insert a dummy record with amount zero.
+
+This jobs requires the `TransformedSpendData` data file in blob storage, which is generated by a pipeline job in Azure Data Factory.
+The output is stored on blob storage in folder  `SpendDataFilledMissingMonth`.
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.main
```
Comment (charlietw) [inline README.md]: Worth adding to the `Makefile` and amending to `make pipeline_local` or similar?

File: README.md [hunk 4]
```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
...
+
+Note: For the moment, this job only run forecast for several pre-defined combinations, due to the long computation time needed for running SARIMA model (around 10 minutes per each combination).
+This behaviour can be changed easily in the related python file
+
+This jobs requires the `SpendDataFilledMissingMonth` data file in blob storage, which is generated by the job above.
+The output is stored on blob storage in folder `ForecastOutput`.
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.jobs.run_forecast_and_output_to_blob
```
Comment (charlietw) [inline README.md]: Same here with `Makefile`

File: README.md [hunk 5]
```diff
@@ -100,17 +102,89 @@ Below are the steps for running and performing data analysis on your local machi
...
+
+
+This jobs requires the 
+`Customers`, `FrameworkCategoryPillar`, `SpendAggregated` data file in blob storage, which are pulled from the tables under the same name in SQL Server.
+There is a pipeline job in Azure Data Factory to pull those file from SQL server to blob storage.
+The output is stored on blob storage in folder `ActiveCombinations`.
+
+##### To run this job on a local machine:
+```shell
+poetry run python -m pipeline.jobs.get_active_combinations
```
Comment (charlietw) [inline README.md]: Same here with `Makefile`

File: test/get_active_combinations_test.py [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+        ("Digital Future", "Unassigned"),
+    ]
+
+    actual = get_active_combinations(
+        customers=customers,
+        framework_category_pillar=framework_category_pillar,
+        spend_aggregated=spend_aggregated,
+    )
+
+    assert actual == expected
```
Comment (charlietw) [inline test/get_active_combinations_test.py:77]: Great test!

**Review by charlietw — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (kapppa-joe) [reply README.md]: Thanks! Updated this to `make unit_testing`

**Review by charlietw — APPROVED** [review]

**Pull request merged**